packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -145,66 +168,407 @@ String getGradleVersionForAndroidPlugin(Directory directory, Logger logger) {
   return getGradleVersionFor(androidPluginVersion ?? 'unknown');
 }
 
-/// Returns true if [targetVersion] is within the range [min] and [max] inclusive.
-bool _isWithinVersionRange(
-  String targetVersion, {
-  required String min,
-  required String max,
-}) {
-  final Version? parsedTargetVersion = Version.parse(targetVersion);
-  final Version? minVersion = Version.parse(min);
-  final Version? maxVersion = Version.parse(max);
-  return minVersion != null &&
-      maxVersion != null &&
-      parsedTargetVersion != null &&
-      parsedTargetVersion >= minVersion &&
-      parsedTargetVersion <= maxVersion;
+/// Returns either the gradle-wrapper.properties value from the passed in
+/// [directory] or if not present the version available in local path.
+///
+/// If gradle version is not found null is returned.
+/// [directory] should be and android directory with a build.gradle file.
+Future<String?> getGradleVersion(
+    Directory directory, Logger logger, ProcessManager processManager) async {
+  final File propertiesFile = directory
+      .childDirectory(gradleDirectoryName)
+      .childDirectory(gradleWrapperDirectoryName)
+      .childFile(gradleWrapperPropertiesFilename);
+
+  if (propertiesFile.existsSync()) {
+    final String wrapperFileContent = propertiesFile.readAsStringSync();
+
+    // Expected content format (with lines above and below).
+    // Version can have 2 or 3 numbers.
+    // 'distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-all.zip'
+    final RegExp distributionUrlRegex = RegExp(r'distributionUrl=.*\.zip');
+
+    final RegExpMatch? distributionUrl =
+        distributionUrlRegex.firstMatch(wrapperFileContent);
+    if (distributionUrl != null) {
+      // Expected content: 'gradle-7.4.2-all.zip'
+      final String? gradleZip = distributionUrl.group(0);
+      if (gradleZip != null) {
+        final String gradleVersion = gradleZip.split('-')[1];
```

Comment:
We should probably check the length before indexing into it. If someone decides that, say, starting with gradle 8 it should be `gradle_8.x.y_all.zip`, we don't want the tool to crash with an index error.

Reply:
Updated and test added to expect null with the same logged warning as the other parse failures.